daemons/controld/controld_schedulerd.c

```diff
@@ -438,11 +438,20 @@ force_local_option(xmlNode *xml, const char *attr_name, const char *attr_value)
     freeXpathObject(xpathObj);
 }
 
+static gboolean
+sleep_timer(gpointer data)
+{
+    controld_set_fsa_action_flags(A_PE_INVOKE);
+    controld_trigger_fsa();
+    return G_SOURCE_REMOVE;
+}
+
 static void
 do_pe_invoke_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void *user_data)
 {
     char *ref = NULL;
     pid_t watchdog = pcmk__locate_sbd();
+    mainloop_timer_t *timer = NULL;
```

Comment:
One thing I forgot to mention earlier... we also need to clean up this timer when we're done with it, which means in the `sleep_timer` function we need to call `mainloop_timer_del` on the timer object.  But in order to do that, it needs to be visible.

So, this should be a static variable somewhere towards the top of the file with a more descriptive name - maybe `cib_retry_timer`, then call `mainloop_timer_del` before returning from `sleep_timer`.